shell/browser/api/electron_api_session.cc

```diff
@@ -427,14 +427,36 @@ v8::Local<v8::Promise> Session::SetProxy(gin::Arguments* args) {
     return handle;
   }
 
-  std::string proxy_rules, bypass_list, pac_url;
+  std::string mode, proxy_rules, bypass_list, pac_url;
 
   options.Get("pacScript", &pac_url);
   options.Get("proxyRules", &proxy_rules);
   options.Get("proxyBypassRules", &bypass_list);
 
+  if (!options.Get("mode", &mode)) {
+    if (!pac_url.empty()) {
+      mode = "pac_script";
+    } else {
+      mode = "fixed_servers";
+    }
+  }
   // pacScript takes precedence over proxyRules.
-  if (!pac_url.empty()) {
+  if (mode == "direct") {
```

Comment:
Can you use [StringToProxyMode](https://source.chromium.org/chromium/chromium/src/+/master:components/proxy_config/proxy_prefs.h;l=65) util and compare against the `ProxyPrefs::ProxyMode` , it will guarantee a level of upstream api breakages wrt the strings used.

Reply:
ok.